app/models/classification.rb

```diff
@@ -10,11 +10,7 @@ class Classification < ApplicationRecord
   before_destroy :delete_tags_and_entries
 
   validates :description, :presence => true, :length => {:maximum => 255}
-  validates :description, :uniqueness => {:scope => [:parent_id]}, :if => proc { |c|
-    cond = c.class.in_region(region_id).where(:parent_id => c.parent_id, :description => c.description)
-    cond = cond.where.not(:id => c.id) unless c.new_record?
-    cond.exists?
-  }
+  validates :description, :unique_within_region => {:scope => :parent_id}, :if => :description_changed?
```

Comment:
I don't understand the `description_changed?` part here....is that an optimization?

Reply:
yes, this runs on record changes (because the description changed
if the description does not change / stays blank - the previous line will kick in and complain